packages/features/data-table/DataTableProvider.tsx

```diff
@@ -184,6 +183,18 @@ export function DataTableProvider({
     setPageIndex,
   });
 
+  const clearAll = useCallback(
```

Comment:
I moved this function to here, with almost no change.

One change: I added `setSegmentIdAndSaveToLocalStorage` as the useCallback dependency.